lib/Interpreter/InterOp.cpp

```diff
@@ -2165,5 +2165,28 @@ namespace InterOp {
 
     return names;
   }
+
+  std::string GetFunctionArgDefault(TCppFunction_t func, TCppIndex_t param_index)
+  {
+    auto *D = (clang::Decl *)func;
+    auto *FD = llvm::dyn_cast_or_null<clang::FunctionDecl>(D);
+    auto PI = FD->getParamDecl(param_index);
+
+    if (PI->hasDefaultArg())
+    {
+      std::string Result;
+      llvm::raw_string_ostream OS(Result);
+      Expr *DefaultArgExpr = const_cast<Expr *>(PI->getDefaultArg());
+      DefaultArgExpr->printPretty(OS, nullptr, PrintingPolicy(LangOptions()));
+
+      if (!Result.empty() && Result.back() == '.')
+      {
+        Result += '0';
+      }
```

Comment:
- Actually the main reason to add this branch was that my modified code was giving output for float type default argument as **4.** (rather than **4.0**) , so in order to get proper output I added this branch to check that if default argument is float type and if it stores value like "3.,4.,5." then the code makes sure to add "0" at end of returned string 
- I hope I have justified the question asked by you correctly
- If there are still any questions I am happy to explain you further about it

Reply:
We need to use a proper pretty printing. This fixes the test case and not the underlying issue.

Reply:
@vgvassilev I think we can just remove the `0` from the test because I don't see any reason that we would need it in the output. Do you have a case where it could be problematic?

Reply:
> @vgvassilev I think we can just remove the `0` from the test because I don't see any reason that we would need it in the output. Do you have a case where it could be problematic?

I'd be fine with that if we add a test case with a default argument `3.123` and it passes.